lore/io/__init__.py

```diff
@@ -138,11 +139,37 @@ def download(remote_url, local_path=None, cache=True, extract=False):
 
     else:
         local_path = temp_path
-        
     return local_path
 
 
-def upload(local_path, remote_path=None):
+# Note: This can be rewritten in a more efficient way
+# https://stackoverflow.com/questions/11426560/amazon-s3-boto-how-to-delete-folder
+def delete_folder(remote_url):
```

Comment:
What do you think about making a single `delete` with `recursive=False` default, and `recursive=True` option to handle folders?

Reply:
Good idea. Done!